### PR TITLE
Unit-tests improvements

### PR DESCRIPTION
**Pull request by bvfalcon**
With [PR 8084](https://github.com/apache/camel/pull/8084) 3 errors comes in unit-tests for camel-jetty component, they are fixed in this PR.
And this PR contains small unit-test improvement in different components: instead of "success" execution of some unit-tests they are marked as "skipped" (using JUnit5 Assumptions mechanism)

### CODE REVIEW

**Review by davsclaus — APPROVED** [review]

**Review by orpiske — APPROVED** [review]
Thanks for your contribution

File: components/camel-salesforce/camel-salesforce-component/src/test/java/org/apache/camel/component/salesforce/CompositeApiIntegrationTest.java [hunk 1]
```diff
@@ -206,9 +208,7 @@ public void shouldSupportQueryAll() {
 
     @Test
     public void shouldSupportRelatedObjectRetrieval() {
-        if (Version.create(version).compareTo(Version.create("36.0")) < 0) {
-            return;
-        }
+        assumeFalse(Version.create(version).compareTo(Version.create("36.0")) < 0, "Version must be less than 36.0");
```
Comment (essobedo) [inline components/camel-salesforce/camel-salesforce-component/src/test/java/org/apache/camel/component/salesforce/CompositeApiIntegrationTest.java:211]:
```suggestion
        assumeFalse(Version.create(version).compareTo(Version.create("36.0")) < 0, "Version must be greater than or equal to 36.0");
```

File: components/camel-xmlsecurity/src/test/java/org/apache/camel/dataformat/xmlsecurity/EncryptionAlgorithmTest.java [hunk 2]
```diff
@@ -197,6 +193,7 @@ public void configure() {
 
     @Test
     public void testAES256GCM() throws Exception {
+        assumeTrue(TestHelper.UNRESTRICTED_POLICIES_INSTALLED, "Test preconditions failed: UNRESTRICTED_POLICIES_INSTALLED="+TestHelper.UNRESTRICTED_POLICIES_INSTALLED);
         if (!TestHelper.UNRESTRICTED_POLICIES_INSTALLED) {
             return;
         }
```
Comment (essobedo) [inline components/camel-xmlsecurity/src/test/java/org/apache/camel/dataformat/xmlsecurity/EncryptionAlgorithmTest.java:199]: remove me

Comment (essobedo) [reply components/camel-xmlsecurity/src/test/java/org/apache/camel/dataformat/xmlsecurity/EncryptionAlgorithmTest.java:199]: Missing fix for https://github.com/apache/camel/blob/ea4bf39a9d6dd35eebe1c2af60e8e05b0a4fce02/components/camel-xmlsecurity/src/test/java/org/apache/camel/dataformat/xmlsecurity/EncryptionAlgorithmTest.java#L297-L299 and https://github.com/apache/camel/blob/ea4bf39a9d6dd35eebe1c2af60e8e05b0a4fce02/components/camel-xmlsecurity/src/test/java/org/apache/camel/dataformat/xmlsecurity/EncryptionAlgorithmTest.java#L325-L327

File: core/camel-management/src/test/java/org/apache/camel/management/ManagedPooledExchangeTest.java [hunk 3]
```diff
@@ -59,9 +60,7 @@ protected CamelContext createCamelContext() throws Exception {
     @Test
     public void testSameExchange() throws Exception {
         // JMX tests dont work well on AIX CI servers (hangs them)
-        if (isPlatform("aix")) {
-            return;
-        }
+        assumeFalse(isPlatform("aix"));
```
Comment (essobedo) [inline core/camel-management/src/test/java/org/apache/camel/management/ManagedPooledExchangeTest.java:63]: I'm wondering if it is needed as the class is annotated with `@DisabledOnOs(OS.AIX)`

File: components/camel-jetty/src/test/java/org/apache/camel/component/jetty/HttpsAsyncRouteTest.java [hunk 4]
```diff
@@ -94,9 +95,7 @@ protected void restoreSystemProperties() {
     @Test
     public void testEndpoint() throws Exception {
         // these tests does not run well on Windows
-        if (isPlatform("windows")) {
-            return;
-        }
+        assumeFalse(isPlatform("windows"), "Test is not intended for windows");
```
Comment (essobedo) [inline components/camel-jetty/src/test/java/org/apache/camel/component/jetty/HttpsAsyncRouteTest.java:98]: Maybe we could annotate the test with `@DisabledOnOs(OS.WINDOWS)`instead?

File: components/camel-jetty/src/test/java/org/apache/camel/component/jetty/HttpsAsyncRouteTest.java [hunk 5]
```diff
@@ -125,9 +124,7 @@ public void testEndpoint() throws Exception {
     @Test
     public void testEndpointWithoutHttps() {
         // these tests does not run well on Windows
-        if (isPlatform("windows")) {
-            return;
-        }
+        assumeFalse(isPlatform("windows"), "Test is not intended for windows");
```
Comment (essobedo) [inline components/camel-jetty/src/test/java/org/apache/camel/component/jetty/HttpsAsyncRouteTest.java:127]: ditto

File: core/camel-core/src/test/java/org/apache/camel/processor/SplitterWithScannerIoExceptionTest.java [hunk 6]
```diff
@@ -20,13 +20,14 @@
...
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+
 public class SplitterWithScannerIoExceptionTest extends ContextTestSupport {
 
     @Test
     public void testSplitterStreamingWithError() throws Exception {
-        if (isPlatform("aix") || isJavaVendor("ibm")) {
-            return;
-        }
+        assumeFalse(isPlatform("aix"));
```
Comment (essobedo) [inline core/camel-core/src/test/java/org/apache/camel/processor/SplitterWithScannerIoExceptionTest.java:29]: Maybe we could annotate the test with `@DisabledOnOs(OS.AIX)` instead?

File: core/camel-management/src/test/java/org/apache/camel/management/ManagedRouteStopWithAbortAfterTimeoutTest.java [hunk 7]
```diff
@@ -33,16 +33,16 @@
...
 @DisabledOnOs(OS.AIX)
 public class ManagedRouteStopWithAbortAfterTimeoutTest extends ManagementTestSupport {
 
     @Test
     public void testStopRouteWithAbortAfterTimeoutTrue() throws Exception {
         // JMX tests dont work well on AIX or windows CI servers (hangs them)
-        if (isPlatform("aix") || isPlatform("windows")) {
-            return;
-        }
+        assumeFalse(isPlatform("aix"));
```
Comment (essobedo) [inline core/camel-management/src/test/java/org/apache/camel/management/ManagedRouteStopWithAbortAfterTimeoutTest.java:44]: Should not be needed as the class is annotated with @DisabledOnOs(OS.AIX)

File: core/camel-management/src/test/java/org/apache/camel/management/ManagedRouteStopWithAbortAfterTimeoutTest.java [hunk 8]
```diff
@@ -33,16 +33,16 @@
...
+        assumeFalse(isPlatform("windows"));
```
Comment (essobedo) [inline core/camel-management/src/test/java/org/apache/camel/management/ManagedRouteStopWithAbortAfterTimeoutTest.java:45]: Maybe we could annotate the test with `@DisabledOnOs(OS.WINDOWS)`instead?

File: core/camel-management/src/test/java/org/apache/camel/management/ManagedRouteStopWithAbortAfterTimeoutTest.java [hunk 9]
```diff
@@ -82,9 +82,8 @@ public void testStopRouteWithAbortAfterTimeoutTrue() throws Exception {
     @Test
     public void testStopRouteWithAbortAfterTimeoutFalse() throws Exception {
         // JMX tests dont work well on AIX or windows CI servers (hangs them)
-        if (isPlatform("aix") || isPlatform("windows")) {
-            return;
-        }
+        assumeFalse(isPlatform("aix"));
+        assumeFalse(isPlatform("windows"));
```
Comment (essobedo) [inline core/camel-management/src/test/java/org/apache/camel/management/ManagedRouteStopWithAbortAfterTimeoutTest.java:86]: ditto

File: core/camel-management/src/test/java/org/apache/camel/management/ManagedThrottlerTest.java [hunk 10]
```diff
@@ -113,10 +113,7 @@ public void testManageThrottler() throws Exception {
     public void testThrottleVisableViaJmx() throws Exception {
         // JMX tests dont work well on AIX CI servers (hangs them)
         assumeFalse(isPlatform("aix"));
-        if (isPlatform("windows")) {
-            // windows needs more sleep to read updated jmx values so we skip as we dont want further delays in core tests
-            return;
-        }
+        assumeFalse(isPlatform("windows"));
```
Comment (essobedo) [inline core/camel-management/src/test/java/org/apache/camel/management/ManagedThrottlerTest.java:116]: Maybe we could annotate the test with `@DisabledOnOs(OS.WINDOWS)`instead?

**Review by essobedo — COMMENTED** [review]
Thx for the contribution, here are my few remarks

File: core/camel-management/src/test/java/org/apache/camel/management/ManagedPooledExchangeTest.java (hunk 3 above)
Comment (bvfalcon) [reply core/camel-management/src/test/java/org/apache/camel/management/ManagedPooledExchangeTest.java:63]: Excellent note! This annotation is much better, than assumption. I replace it there possible

File: core/camel-core/src/test/java/org/apache/camel/processor/ThrottlerTest.java [hunk 11]
```diff
@@ -24,24 +24,19 @@
...
     private static final int INTERVAL = 500;
     private static final int TOLERANCE = 50;
     private static final int MESSAGE_COUNT = 9;
 
-    protected boolean canTest() {
-        // skip test on windows as it does not run well there
-        return !isPlatform("windows");
-    }
-
+    @DisabledOnOs(OS.WINDOWS)
```
Comment (essobedo) [inline core/camel-core/src/test/java/org/apache/camel/processor/ThrottlerTest.java:37]: As all tests seem to be disabled for windows, maybe we could add the annotation only once at the class definition level?

**Review by essobedo — COMMENTED** [review]
Much better 👍 , only one last small remark to go

File: core/camel-core/src/test/java/org/apache/camel/processor/ThrottlerTest.java (hunk 11 above)
Comment (bvfalcon) [reply core/camel-core/src/test/java/org/apache/camel/processor/ThrottlerTest.java]: Change for this and some other cases

File: components/camel-quartz/src/test/java/org/apache/camel/component/quartz/SpringQuartzPersistentStoreRestartRouteTest.java [hunk 12]
```diff
@@ -1,65 +1,62 @@
...
-        mock.reset();
-        mock.expectedMinimumMessageCount(2);
-
-        context().getRouteController().startRoute("myRoute");
-
-        assertMockEndpointsSatisfied();
-    }
-
-}
+/*
```
Comment (essobedo) [inline components/camel-quartz/src/test/java/org/apache/camel/component/quartz/SpringQuartzPersistentStoreRestartRouteTest.java:1]: Any idea why it is seen as the full change?

Comment (bvfalcon) [reply components/camel-quartz/src/test/java/org/apache/camel/component/quartz/SpringQuartzPersistentStoreRestartRouteTest.java:1]: 2 ideas: encoding or/and line endings

**Review by essobedo — APPROVED** [review]
LGTM 👍

**Pull request merged**